src/einsteinpy/metric/schwarzschild.py

```diff
@@ -148,53 +111,58 @@ def calculate_trajectory(
 
         Returns
         -------
-        tuple
-            (~numpy.array of lambda, (n,8) shape ~numpy.array of [t, pos1, pos2, pos3, velocity_of_time, vel1, vel2, vel3])
+        ~numpy.ndarray
+            N-element array containing proper time.
+        ~numpy.ndarray
+            (n,8) shape array of [t, x1, x2, x3, velocity_of_time, v1, v2, v3] for each proper time(lambda).
 
         """
-        vec_list = list()
-        lambda_list = list()
-        singularity_reached = False
         ODE = RK45(
             fun=self.f_vec,
             t0=start_lambda,
             y0=self.initial_vec,
             t_bound=end_lambda,
             **OdeMethodKwargs
         )
-        _scr = self.schwarzschild_r.value * 1.001
+
+        vecs = list()
+        lambdas = list()
+        singularity_reached = False
+        _scr = self.scr.value * 1.001
+
         while ODE.t < end_lambda:
-            vec_list.append(ODE.y)
-            lambda_list.append(ODE.t)
+            vecs.append(ODE.y)
+            lambdas.append(ODE.t)
             ODE.step()
             if (not singularity_reached) and (ODE.y[1] <= _scr):
-                warnings.warn(
-                    "r component of position vector reached Schwarzchild Radius. ",
-                    RuntimeWarning,
-                )
+                warnings.warn("particle reached Schwarzchild Radius. ", RuntimeWarning)
                 if stop_on_singularity:
                     break
                 else:
                     singularity_reached = True
 
-        def _not_cartesian():
-            return (np.array(lambda_list), np.array(vec_list))
-
-        def _cartesian():
-            self.units_list = [
-                u.s,
-                u.m,
-                u.m,
-                u.m,
-                u.one,
-                u.m / u.s,
-                u.m / u.s,
-                u.m / u.s,
-            ]
-            return (np.array(lambda_list), S2C_8dim(np.array(vec_list)))
-
-        choice_dict = {0: _not_cartesian, 1: _cartesian}
-        return choice_dict[int(return_cartesian)]()
+        vecs, lambdas = np.array(vecs), np.array(lambdas)
+
+        if not return_cartesian:
+            returnval = (lambdas, vecs)
+        else:
+            sph_vecs = list()
+            for v in vecs:
+                si_vals = (
+                    SphericalDifferential(
+                        v[1] * u.m,
+                        v[2] * u.rad,
+                        v[3] * u.rad,
+                        v[5] * u.m / u.s,
+                        v[6] * u.rad / u.s,
+                        v[7] * u.rad / u.s,
+                    )
+                    .cartesian_differential()
+                    .si_values()
+                )
+                sph_vecs.append(np.hstack((v[0], si_vals[:3], v[4], si_vals[3:])))
+            returnval = (lambdas, np.array(sph_vecs))
```

Comment:
```suggestion
            return lambdas, np.array(sph_vecs)
```

Reply:
I'm feeling very sad to say that our dear codefactor cries on doing this. 😆

Reply:
Let it cry!